docs/DeveloperGuide.md

```diff
@@ -1,38 +1,426 @@
 # Developer Guide
-
+<!----------------------------Table of Contents----------------------------------------->
+- [Acknowledgements](#acknowledgements)
+- [Design](#design)
+  - [Architecture](#architecture)
+  - [UI Component](#ui-component)
+  - [Logic Component](#logic-component)
+  - [Model Component](#model-component)
+  - [Storage Component](#storage-component)
+  - [Common classes](#common-classes)
+- [Implementation](#implementation)
+  - [Add Module](#add-module)
+  - [Remove Module](#remove-module)
+  - [Mark Module As Taken](#mark-module-as-taken)
+  - [List Modules](#list-modules)
+  - [Display Status](#display-status)
+  - [Get Module Details](#get-module-details)
+  - [Initialise User](#initialise-user)
+  - [Save To Local Drive](#save-plan-to-local-drive)
+- [Documentation, logging, testing, configuration, dev-ops](#documentation-logging-testing-configuration-dev-ops)
+  - [Documentation](#documentation)
+  - [Logging](#logging)
+  - [Testing](#testing)
+  - [Configuration](#configuration)
+  - [Dev-ops](#dev-ops)
+- [Appendix A: Requirements](#appendix-a--requirements)
+- [Appendix B: Instructions for manual testing](#appendix-b--instructions-for-manual-testing)
+<!----------------------------Acknowledgements----------------------------------------->
 ## Acknowledgements
+References made from [AddressBook 2](https://github.com/se-edu/addressbook-level2) and [AddressBook 3](https://github.com/se-edu/addressbook-level3).
+
+
+<!----------------------------Design----------------------------------------->
+## Design
+
+### Architecture
+
+![ArchitectureDiagram](uml/diagrams/Architecture.png)
+
+### UI Component
+![UIClassDiagram](uml/diagrams/UiClass.png)
```

Comment:
Perhaps the UI role is redundant? If roles are used maybe moving closer to the side of the class it references would be better than leaving it in the center and creating ambiguity as to which class has that role?
![image](https://user-images.githubusercontent.com/88386360/228744469-0848cb17-637c-441d-9649-01b7e3740019.png)